executor/update.go

```diff
@@ -263,6 +264,12 @@ func (e *UpdateExec) updateRows(ctx context.Context) (int, error) {
 				txn.GetSnapshot().SetOption(kv.CollectRuntimeStats, e.stats.SnapshotRuntimeStats)
 			}
 		}
+		if config.TopSQLEnabled() {
+			txn, err := e.ctx.Txn(false)
```

Comment:
It should be `e.ctx.Txn(true)` here.
To call  `SetOption()`, the underlying `kv.Transaction` must be actived.